src/OlStyler.js

```diff
@@ -70,6 +70,52 @@ function hexToRGB(hex, alpha) {
 }
 
 function polygonStyle(style) {
+
+  if (style.fill
+        && style.fill.graphicfill
+        && style.fill.graphicfill.graphic
+        && style.fill.graphicfill.graphic.externalgraphic
+        && style.fill.graphicfill.graphic.externalgraphic.onlineresource) {
+    // Check symbolizer metadata to see if the image has already been loaded.
+    switch (style.__loadingState) {
+      case IMAGE_LOADED:
+        const { image, width, height } = imageCache[style.fill.graphicfill.graphic.externalgraphic.onlineresource]
+        const maxSide = Math.max(width, height)
+
+        let imageRatio = 1
+        if (style.fill.graphicfill.graphic.size && maxSide != style.fill.graphicfill.graphic.size) {
+            imageRatio = style.fill.graphicfill.graphic.size / maxSide
+        }
+        let cnv = document.createElement('canvas');
+        let ctx = cnv.getContext('2d')
+        let pattern
+        if (imageRatio == 1) {
+            pattern = ctx.createPattern(image, 'repeat')
+        } else {
+            let tempCanvas = document.createElement("canvas")
+            let tCtx = tempCanvas.getContext("2d")
+
+            tempCanvas.width = width * imageRatio
+            tempCanvas.height = height * imageRatio
+            tCtx.drawImage(image, 0, 0, width, height, 0, 0, width * imageRatio, height * imageRatio)
+            pattern = ctx.createPattern(tempCanvas, 'repeat')
+        }
+        return(new Style({
+            fill: new Fill({
+              color: pattern
+            })
+          }));
+
+      case IMAGE_LOADING:
+        return imageLoadingStyle;
```

Comment:
It would be better to define and use a separate polygonImageLoadingStyle here, because imageLoadingStyle only works for points. You can add a new one next to imageLoadingStyle. Maybe also rename imageLoadingStyle to pointImageLoadingStyle for clarity?

Reply:
You are right. I added polygon styles